components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -232,59 +321,133 @@ impl IndexScanExecutorImpl {
                 return Err(other_err!("Unexpected missing column {}", col_id));
             }
         }
+        Ok(())
+    }
 
-        if self.decode_handle {
-            // For normal index, it is placed at the end and any columns prior to it are
-            // ensured to be interested. For unique index, it is placed in the value.
+    fn extract_columns_from_datum_format(
+        datum: &mut &[u8],
+        columns: &mut [LazyBatchColumn],
+    ) -> Result<()> {
+        for column in columns {
+            if datum.is_empty() {
+                return Err(other_err!("Value is missing some columns"));
```

Comment:
Please give information about the column id to make it easy to debug when problem occurs.